libraries/standard/mqtt/src/mqtt.c

```diff
@@ -315,6 +315,8 @@ static int32_t sendPacket( MQTTContext_t * pContext,
 
     assert( pContext != NULL );
     assert( pContext->callbacks.getTime != NULL );
+    assert( pContext->transportInterface.send != NULL );
+    assert( pIndex != NULL );
```

Comment:
Do we expect sendPacket() to always be called with `bytesToSend` > 0? If so, should we assert that here? If not, should we change line 357 so that the packet send time is not updated if nothing was sent?

Reply:
I believe it is a better to not assume that the bytesToSend will always be greater than zero. It is my mistake for changing 357, i believe it was `if( totalBytesSent > 0 )` in before my last commit. Thanks for catching.